### PR TITLE
Enable EIP 1559 on ropsten

### LINKED ISSUE #12116: Allow users to customize priority fee
As as user
I want to be able to customize the priority fee
So that I can make sure my transaction is processed in a timely fashion

### Description

When sending a transaction, we want to allow the user to set the priority fee.

Because eip 1559 will change the way that fees are handled, this screen will eventually replace the current existing one.

https://www.figma.com/file/XK7vrp3pZdQ5gqQVJ7Pqgz/EIP-1559-Presentation?node-id=26%3A1793

Depending on the amount of work the slider and/or error conditions can be implemented separately.

### Technical implementation

The screen should not replace the current existing screen, but should be a toggle (the toggle mechanism can be implemented in a separate PR). 
This is because we only want to go live once the network is ready.

Currently it's not completely clear how we will be pulling in the data about base fee/maximum fee etc, so given that we have a hard deadline for this task, we can go ahead an implement the frontend separately and use mock data until things are clearer.

### PR DESCRIPTION
**Pull request by cammellos**
Enable EIP 1559 on ropsten network.

This might require e2e test changes.

### Testing

If you set custom fee when sending a transaction, you should see the new interface on ropsten.

Fixes: #12116

status: ready

### CODE REVIEW

**Review by flexsurfer — APPROVED** [review]

**Comment by churik** [comment]
@cammellos 
maybe I did smth in a wrong way, but I see old interface on Ropsten if I recover multiaccount from seed on builds with https://github.com/status-im/status-react/commit/f4f97ae50bd9f1d63031e52c3771da745db57183

https://user-images.githubusercontent.com/4557972/127132264-98659a06-f4e3-4843-b79c-fb461e85de4d.mp4

I couldn't find any related setting in Advanced.

**Comment by cammellos** [comment]
@churik sorry! I enabled it on dev, but I need to enable it on PRs as well, let me fix that, I'll let you know once I verify it.

**Comment by churik** [comment]
@cammellos @rasom 
at attempt to set custom network fee on Ropsten network got error: 
IMAGE 2021-08-02 11:39:19
Logs: 
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/6915970/Status-debug-logs.zip)

**Comment by rasom** [comment]
@churik that issue is fixed

**Comment by churik** [comment]
can't sign transaction on ropsten - all failed with reason `intrisic gas too low`
IMAGE 2021-08-04 14:01:36

**Comment by churik** [comment]
@rasom 
### ISSUE 1: Can't sign transaction with default gas price on Ropsten
Build with commit https://github.com/status-im/status-react/commit/1b2f80f0f51e435f34cd19bbfc10b8ea50a7b2d4
IMAGE 2021-08-05 12:41:01
IMAGE 2021-08-05 12:41:05
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/6937974/Status-debug-logs.zip)


### ISSUE 2: network fee is not calculated during some time after navigating to "Sending" screen
Tested on build with commit https://github.com/status-im/status-react/commit/8c89720ce38a1801ee53a1f706c04b70003fc4a2

**The easiest way to reproduce:**
1) switch to Ropsten
2) fill amount and address with valid data
3) Tap on sign 
4) On "sending" screen tap cancel
5) try to start again

**Expected result:**
network fee is calculated

**Actual result:**
network fee is not calculated anymore during some time; 

https://user-images.githubusercontent.com/4557972/128339813-d8c5806a-36d5-4c32-be96-ae8313e9a577.mp4

**Comment by churik** [comment]
### ISSUE 3: can't sign transaction with keycard
Tried on both mainnet and ropsten
**Steps:**
1) restore multiaccount from seed phrase with keycard
2) try to sign transaction


**Expected result:**
tx is signed
**Actual result:**
nothing happened, processing
**OS:** IOS, Android
[status_logs.zip](https://github.com/status-im/status-react/files/6939695/status_logs.zip)

**Comment by churik** [comment]
@rasom on last build with https://github.com/status-im/status-react/commit/f75d5f6f54ffc734eaace5d1c0f04797ffc7ba34 suggestion fee is 0 on Ropsten

**Comment by churik** [comment]
@rasom thank you, keycard issue is fixed!
Meanwhile found another issue

### ISSUE 4: can't request STT or send any tx in simpledapp.eth

**Steps:**
1) restore multiaccount from seed phrase (with ETHro)
2) switch to Ropsten
3) open simpledapp.eth
4) Assets > Request STT

**Expected result:**
can sign tx

**Actual result:**
IMAGE 2021-08-10 11:53:24

**OS:** IOS, Android

**Comment by rasom** [comment]
@churik issue 4 is fixed

**Comment by Serhy** [comment]
**Issue 5**
App crashed when opening Network Fee bottom sheet when: a) sending transaction in 1-1 chat b) sending transaction in Dapp  

**Reproduction 1:**
1) Recover account with ETHro
2) Join 1-1 chat 
3) Send transaction in 1-1 chat and open Network Fee dialog before signing it
Actual result: https://saucelabs.com/jobs/e54752589a764ca8ac904b1334e06a61?auth=1fcd0ee8ff8f39ac9ffaadbfa9e12421
Screenshot 2021-08-11 at 20 52 49

**Reproduction 2:**
1) Recover account with ETHro
2) Open simpledapp.eth
3) `Assets` -> `Request STT` -> tap `Network Fee`

**Issue 6**
Is it possible to implement scroll in Max Priority Fee bottom sheet? In a case user wants to edit Gas amount - the field it hidden at the top of screen and typing possible in blind


https://user-images.githubusercontent.com/8749671/129079839-47d8ed93-70a4-49a3-8fb6-de599f0c1808.mp4

**Comment by churik** [comment]
Umbrella comment for issues (to be updated):

- [x] ISSUE 1: Can't sign transaction with default gas price on Ropsten
- [x] ISSUE 2: network fee is not calculated during some time after navigating to "Sending" screen
- [x] ISSUE 3: can't sign transaction with keycard
- [x] ISSUE 4: can't request STT or send any tx in simpledapp.eth
- [x] ISSUE 5: App crashed when opening Network Fee bottom sheet when: a) sending transaction in 1-1 chat b) sending transaction in Dapp
- [ ] ISSUE 6: scroll in Max Priority Fee bottom sheet: will be handled separately, as doesn't relateto PR
- [x] ISSUE 7: Set max option doesn't set any value for ETH
- [ ] ISSUE 8: Sometimes txs are stuck in "Pending" state in tx history: couldn't find a way to reproduce consistently
- [x] ISSUE 9: network fee is set to 0 if using Set max on ETH
- [x] ISSUE 10: Transaction underpriced at attempt to deploy contract from Dapp
- [x] ISSUE 11: No basic validation for custom network fee
- [x] ISSUE 12: Transaction underpriced at attempt to send assets in 1-1 chat from keycard multiaccount (MAINNET)
- [x] ISSUE 13: On the custom fee screen "Save" button is still available even user sets tip o invalid params (let's say, 0 or non-numeric)
- [x] ISSUE 14: On dap.ps on "Upvote" button tip limit / price limit is not set

**Comment by churik** [comment]
@rasom 
### ISSUE 4, 5: still error when opening bottom sheet for sending transaction in 1-1 chat or in dapp

**Steps:**
1) restore multiaccount from seed phrase (with ETH on mainnet)
2) start 1-1 chat
3) send some ETH in 1-1 chat
IMAGE 2021-08-16 13:23:12

**OS:** IOS, Android
Reproducible with simpledapp.eth on ropsten

**Comment by churik** [comment]
@rasom 
another error when tapping `network fee`when requesting some assets in simpledapp.eth (Ropsten):
IMAGE 2021-08-16 16:23:31
Same error when tapping `network fee` when sending assets in 1-1 chat

**Comment by churik** [comment]
@rasom issue 6 is confirmed on Android 9 and Android 11, not relevant for IOS.

**Comment by churik** [comment]
### ISSUE 7: `Set max` option doesn't set any value for ETH

**Steps:**
1) restore multiaccount from seed phrase (with ETH)
2) open wallet > send tx
3) tap on `set max`


**Expected result:**
max amount is set (taking into account default network fee)
**Actual result:**
nothing happend
**OS:** IOS, Android

**Comment by churik** [comment]
### ISSUE 8: Sometimes txs are stuck in "Pending" state in  tx history.
Seems that if ix is mined too fast (within 1-2 mins) it can stuck (only assumption though)
IMAGE 2021-08-17 12:45:02
TX 0.01 SALT is stuck in history in "Pending", relogin doesn't help.
Balance was refreshed only after relogin
Logs: 
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/6999274/Status-debug-logs.zip)

**Comment by churik** [comment]
@rasom 
### ISSUE 9: Network fee is set to 0 if using Set max on ETH

**Steps:**
1) restore multiaccount from seed phrase (with assets)
2) tap Send
3) tap Set max



### ISSUE 10: `Transaction underpriced` at attempt to deploy contract from Dapp
e2e test blocker.
https://saucelabs.com/jobs/fc7a789932fc430d9934fd7c96788a2f?auth=c9f75e5b352bb7352639e0d9b329d6c2 
Steps: 
1) restore multiaccount from seed phrase (with assets on Ropsten)
2) open simpledapp.eth > Transactions > Deploy simple contract

### ISSUE 11: No basic validation for custom network fee, can't set Gas amount limit
All fields (Gas amount limit, Per-gas tip limit fee, Per-Gas price limit) are empty when tapping on Ntwork fee, and Save button is available.

I would suggest to have:
1) Gas amount limit, Per-gas tip limit fee, Per-Gas price limit are filled with current values
2) Save button is unavailable at least when fields have non-numeric values or empty.

At attempt to set Gas amount limit (when typing in field) got an error: 


### ISSUE 12: `Transaction underpriced` at attempt to send assets in 1-1 chat from keycard multiaccount (MAINNET)

https://user-images.githubusercontent.com/4557972/129868465-fbbd58bc-af60-4652-87da-33119c068d01.mp4

**Comment by churik** [comment]
Tested (IOS, Android, keycard/not keycard):
- tx from dapp
- all related e2e are fixed anad passed: [results](https://github.com/status-im/status-react/pull/12429#issuecomment-901947465)
- tx from kk account: tokens, eth, 1-1 chat, wallet
- buy sticker flow (didn't sign tx, as it costs > 50 USD), but checked that tip/base fee is set
- buy ENS flow (didn't sign tx, as it costs > 50 USD), but checked that tip/base fee is set
- collectibles are presented
- sending custom tokens

What's left:
### ISSUE 13: On the custom fee screen "Save" button is still available even user is set to invalid params (let's say, 0 or non-numeric).
Therefore, users can try to sign these txs and will get an error i.e. 'Transaction underpriced' 
IMO would be nice to disable this button in such cases, but it is not a blocker for this feature

### ISSUE 14: On dap.ps on "Upvote" button tip limit / price limit is not set
I'm not sure is it dap.ps issue or the issue on our side.
IMAGE 2021-08-19 17:27:46
Didn't try to sign tx, as it costs > 50 USD
All PR issues are in umbrella comment https://github.com/status-im/status-react/pull/12369#issuecomment-899432252

More notes:
- again saw `ISSUE 8: Sometimes txs are stuck in "Pending" state in tx history: couldn't find a way to reproduce consistently` - but couldn't reproduce it consistently (asked for a help, but no luck)
- tx management (alfa) is working on mainnet now  weird
    - on increase gas old tx will be not replaced in our history, so it will stay there forever
    - buttons "Increase" and "Cancel" are not responsible during some time after creating tx (until tx will appear on etherscan)
    So basically to use this feature is painful, as it seems rawer than it was when I tested it several weeks ago (but still working on a very basic level though)

@cammellos would like to know your opinion: smth from mentioned above seems blocker for PR or it is good to go?
Thank you!

**Comment by cammellos** [comment]
> @cammellos would like to know your opinion: smth from mentioned above seems blocker for PR or it is good to go?

@churik they don't seem to be blocker to me, they should mostly affect advanced users, what do you think?

**Comment by churik** [comment]
@rasom fixed it, so I'm retesting and good to go
